### ISSUE TITLE
Obstacle avoidance and follow modes not working correctly

### ISSUE THREAD

**Issue by larrycturner** [elegooofficial/ELEGOO-Smart-Robot-Car-Kit-V4.0#3]
I'm having problems with obstacle avoidance mode. When I select obstacle avoidance on my phone the bot only backs up and never moves forward.  When I tested it the head rotates side to side and I can hear the click of the ultrasonic sensor but it doesn't seem to sense any object.  When I put it in follow mode the bot moves forward and never stops with nothing in front of it. So the problem is with the ultrasonic unit and I would like to know how to put the Arduino program into test mode to see the distance on the serial monitor.

**Comment by OwaisK21** [comment]
same problem did you got any solutions

**Comment by larrycturner** [comment]
Yes, the problem was I had a bad ultrasonic sensor. I contacted Elegoo and they are sending me a new one. I bought an Elegoo Sensor kit a couple of years ago and used the ultrasonic sensor from it. It works perfectly now.
